tests/test_loop.py

```diff
@@ -223,7 +223,8 @@ def test_loop_ping_timeout(self):
         res = mqtt_client.loop(timeout=2 * keep_alive_timeout)
         assert time.monotonic() - start >= 2 * keep_alive_timeout
         assert len(mocket.sent) > 0
-        assert len(res) == 2
+        # assert len(res) == 2
+        assert len(res) == 3  # not sure if 3 is ok
```

Comment:
What happens before your changes is that the the PINGREQ is sent right when `loop()` is called because `self._last_msg_sent_timestamp` is still 0: https://github.com/adafruit/Adafruit_CircuitPython_MiniMQTT/blob/cbe2177efc2837519de8f50b96f9eab5c1f6ca3f/adafruit_minimqtt/adafruit_minimqtt.py#L937-L945

Then after 4 seconds (the value of `keep_alive_timeout` set by the test case) another PINGREQ is sent and then the `loop()`'s loop (haha) breaks out due to the loop timeout.

With your changes, another PINGREQ is sent right at the boundary of 8 seconds when the `loop()` is about to break out, however probably due to the timing changes (being less precise ?) it manages to check the keep alive for one more time and emits one more PINGREQ.

So from the perspective of your changes modifying the assert to use 3 instead of 2 is fine, however the test case should be changed slightly so that the loop timeout is not aligned to the keep alive timeout.